plugins/modules/aci_dhcp_relay_provider.py

```diff
@@ -0,0 +1,357 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) <year>, <Name> (@<github id>)
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: aci_dhcp_relay_provider
+short_description: Manage DHCP relay policy providers.
+description:
+- Manage DHCP relay policy providers (dhcpRsProv) configuration on Cisco ACI fabrics.
+options:
+  tenant:
+    description:
+    - Name of an existing tenant.
+    type: str
+  relay_policy:
+    description:
+    - Name of an existing DHCP rely policy
+    type: str
+    aliases: [ relay_policy_name ]
+  epg_type:
+    description:
+    - Type of EPG the DHCP srver is in
+    type: str
+    choices: [ app_epg, l2_external_net, l3_external_net ]
```

Comment:
UI also allow DN as a type. Should we add that support?

Reply:
I've added it, though I'm not sure if anyone will actually use the raw DN instead of one of the epg_types that calculate the DN for them.